### PR TITLE
feat(fr): add anisama

### PR DESCRIPTION
**Pull request by hollowshiroyuki**
Checklist:

- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [x] Have removed `web_hi_res_512.png` when adding a new extension
- [ ] oil bottle
- [ ] toilet paper
- [ ] 2 sponges
- [ ] ravioli

### CODE REVIEW

File: src/fr/anisama/src/eu/kanade/tachiyomi/animeextension/fr/anisama/AniSama.kt [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+                }
+            }
+        }
+    }
+
+    override fun List<Video>.sort(): List<Video> {
+        val language = preferences.language
+        val quality = preferences.quality
+        val server = preferences.server
+        val serverBlacklist = preferences.serverBlacklist
```
Comment (Secozzi) [inline src/fr/anisama/src/eu/kanade/tachiyomi/animeextension/fr/anisama/AniSama.kt:218]: Maybe move this to videoListParse so it doesn't unnecessary extract videos?

File: src/fr/anisama/src/eu/kanade/tachiyomi/animeextension/fr/anisama/extractors/VidCdnExtractor.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    data class CdnSourceDto(val file: String)
+
+    @Serializable
+    data class CdnResponseDto(val sources: List<CdnSourceDto>)
+
+    fun videosFromUrl(
+        url: String,
+        videoNameGen: (String) -> String = { quality -> quality },
+    ): List<Video> {
+        return runCatching {
```
Comment (Secozzi) [inline src/fr/anisama/src/eu/kanade/tachiyomi/animeextension/fr/anisama/extractors/VidCdnExtractor.kt:31]: Remove runcatching from the extractor, you're catching such errors in the main class anyways

File: src/fr/anisama/src/eu/kanade/tachiyomi/animeextension/fr/anisama/AniSama.kt [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+    override fun searchAnimeSelector() = popularAnimeSelector()
+
+    override fun searchAnimeFromElement(element: Element) = popularAnimeFromElement(element)
+
+    override fun searchAnimeNextPageSelector() = popularAnimeNextPageSelector()
+
+    // =========================== Anime Details ============================
+    private fun Elements.getMeta(name: String) = select(".item:has(.item-title:contains($name)) > .item-content")
+
+    private fun parseStatus(status: String) = when (status) {
```
Comment (Secozzi) [inline src/fr/anisama/src/eu/kanade/tachiyomi/animeextension/fr/anisama/AniSama.kt:125]: Make this a function under `Element` like how you did with `.getMeta()` for consistency's sake

File: src/fr/anisama/src/eu/kanade/tachiyomi/animeextension/fr/anisama/AniSama.kt [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+        val details = document.select(".anime-detail")
+        return SAnime.create().apply {
+            title = details.select(".dynamic-name").text().substringBeforeLast(" ")
+            thumbnail_url = details.select(".film-poster-img").attr("src")
+            url = document.select("link[rel=canonical]").attr("href")
+            artist = details.getMeta("Studio").text()
+            status = parseStatus(details.getMeta("Status").text())
+            description = details.select(".shorting").text()
+            genre = details.getMeta("Genre").text()
+            initialized = true
```
Comment (Secozzi) [inline src/fr/anisama/src/eu/kanade/tachiyomi/animeextension/fr/anisama/AniSama.kt:141]: This line isn't necessary (unless you're overriding `getMangaDetails`, which you aren't)

**Review by Claudemirovsky — APPROVED** [review]
french/10

**Review by Secozzi — APPROVED** [review]

**Comment by hollowshiroyuki** [comment]
damn, was absolutely sure you wouldn't allow that `Triple` atrocity

**Pull request merged**